src/screens/qrScreen/qrScreen.tsx

```diff
@@ -0,0 +1,177 @@
+
+  const [isActive, setIsActive] = useState(true);
+  const [scanResult, setScanResult] = useState(null);
+  const scannerRef = useRef(null);
+  const onSuccess = (e) => {
+    setScanResult(e.data);
+    setIsActive(false);
+  };
+
+  const _handleOpen = () => {
+    console.log('scanResult : ', scanResult);
+    _handleDeepLink(scanResult)
+  };
+
+  const _handleDeepLink = async (url = '') => {
+    if (!url || url.indexOf('ShareMedia://') >= 0) return;
+
+    let routeName;
+    let params;
+    let content;
+    let profile;
+    let keey;
+    
+    const postUrl = postUrlParser(url);
+    const { author, permlink, feedType, tag } = postUrl || {};
+
+    try {
+      if (author) {
+        if (
+          !permlink ||
+          permlink === 'wallet' ||
+          permlink === 'points' ||
+          permlink === 'comments' ||
+          permlink === 'replies' ||
+          permlink === 'posts'
+        ) {
+          let deepLinkFilter;
+          if (permlink) {
+            deepLinkFilter = permlink === 'points' ? 'wallet' : permlink;
+          }
+
+          profile = await getUser(author);
+          routeName = ROUTES.SCREENS.PROFILE;
+          params = {
+            username: get(profile, 'name'),
+            reputation: get(profile, 'reputation'),
+            deepLinkFilter, //TODO: process this in profile screen
+          };
+          keey = get(profile, 'name');
+        } else if (permlink === 'communities') {
+          routeName = ROUTES.SCREENS.WEB_BROWSER;
+          params = {
+            url: url,
+          };
+          keey = 'WebBrowser';
+        } else if (permlink) {
+          content = await getPost(author, permlink, currentAccount.name);
+          routeName = ROUTES.SCREENS.POST;
+          params = {
+            content,
+          };
+          keey = `${author}/${permlink}`;
+        }
+      }
+
+      if (feedType === 'hot' || feedType === 'trending' || feedType === 'created') {
+        if (!tag) {
+          routeName = ROUTES.SCREENS.TAG_RESULT;
+        } else if (/hive-[1-3]\d{4,6}$/.test(tag)) {
+          routeName = ROUTES.SCREENS.COMMUNITY;
+        } else {
+          routeName = ROUTES.SCREENS.TAG_RESULT;
+        }
+        params = {
+          tag,
+          filter: feedType,
+        };
+        keey = `${feedType}/${tag || ''}`;
+      }
+    } catch (error) {
+      // this._handleAlert('deep_link.no_existing_user');
+      console.log('No existing user!');
+    }
+
+    if (!routeName) {
+      const { mode, referredUser } = parseAuthUrl(url);
+      if (mode === 'SIGNUP') {
+        routeName = ROUTES.SCREENS.REGISTER;
+        params = {
+          referredUser,
+        };
+        keey = `${mode}/${referredUser || ''}`;
+      }
+    }
+
+    if (routeName && keey) {
+      navigate({
+        routeName,
+        params,
+        key: keey,
+      });
+    }else{
+      console.log('unable to open url');
```

Comment:
This section can be used to return validity... we can repurpose this method to return routeName, params and key... if these properties exist in returning object, then show validity tick mark otherwise show cross.

Used the same returned data on open button press.

@aliseyalvi this should make more sense, feel free to reach out for any queries
```
const [navData, setNavData] = useState(null);
const [readUrl, setReadUrl] = useState('');
.
.
.

const navData = _handleDeepLink(url);
setIsValid(navData?true:false);
.
.
.
const _openPressed = () => {
    navigate(navData);
}
.
.
.
<>
{
   readUrl && <ValidTick isValid={navData}/>
}
</>
```